Jenkinsfile

```diff
@@ -1,31 +1,9 @@
-#!groovy
-
-def deployBranches = [ "master" ]
-def phase = "verify"
-
-stage ('Build') {
-    node {
-        checkout scm
-        def branch = scm.branches[0].name
-        if (deployBranches.contains(branch)) {
-            phase = "deploy"
-        }
-        echo "Running mvn $phase on branch $branch"
-        sh 'mkdir -p ~/.gnupg'
-        withCredentials([
-            file(credentialsId: 'gpg-pubring', variable: 'GPG_PUB_RING'),
-            file(credentialsId: 'gpg-secring', variable: 'GPG_SEC_RING'),
-            file(credentialsId: 'gradle-settings', variable: 'GRADLE_SETTINGS')]) {
-                try {
-                    sh "./gradlew $phase -P signing.secretKeyRingFile=$GPG_SEC_RING -P extProps=$GRADLE_SETTINGS"
-                } finally {
-                    archiveArtifacts 'build/libs/*.jar'
-                    archiveArtifacts 'build/libs/*.asc'
-                    if (phase == 'deploy') archiveArtifacts 'build/poms/*.xml'
-                    if (phase == 'deploy') archiveArtifacts 'build/poms/*.asc'
-                    junit allowEmptyResults: true, testResults: 'build/test-results/test/*.xml'
-                }
-        }
-        step([$class: 'WsCleanup'])
-    }
-}
\ No newline at end of file
+buildPluginWithGradle(
+   allowEmptyArchive: true,
```

Comment:
```suggestion
```

Reply:
Wait, I'm not sure I understand this suggestion. I thought your change in https://github.com/jenkins-infra/pipeline-library/pull/300 was meant to add an `allowEmptyAchive` parameter (see https://github.com/jenkins-infra/pipeline-library/pull/301). The builds are still failing with the error `No artifacts found that match the file pattern "**/build/libs/*.hpi,**/build/libs/*.jpi". Configuration error?`.

Reply:
Some builds used the cached version of the pipeline library, the last one before your last commits were passing

Reply:
it's not a parameter, I just set it to true always, the previous build passed

Reply:
Ah sorry, I think that the failing build I was looking at must have been using an older version of the pipeline library. I'll drop my PR. Sorry for the confusion!

Reply:
I'm not sure @timja wanted to add a parameter to the pipeline, I think it was only to allow the archive function to skip empty test folders.